### PR TITLE
FEATURE: Improve Media gallery indexer

### PR DESCRIPTION
**Pull request by coderimus**
### Description (*)
Improve the `MediaGalleryUi` indexer approach. Made them extensible. The main goal is to represent the `MediaContent` relation indexer. Also provide some improvements to the general MediaGallery indexer such as time execution and indexer title.

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)
N/A

Depends on: https://github.com/magento/adobe-stock-integration/pull/1157

### CODE REVIEW

File: MediaGalleryUi/Console/Command/IndexAssets.php [hunk 1]
```diff
@@ -33,14 +33,14 @@ class IndexAssets extends Command
...
+     * @param array $indexers
      * @param State $state
      */
     public function __construct(
-        ImagesIndexer $imagesIndexer,
+        array $indexers,
         State $state
     ) {
-        $this->imagesIndexer = $imagesIndexer;
+        $this->indexers= $indexers;
```
Comment (hws47a) [inline MediaGalleryUi/Console/Command/IndexAssets.php]: space is missing here before `=`

File: MediaGalleryUi/Console/Command/IndexAssets.php [hunk 2]
```diff
@@ -51,19 +51,31 @@ public function __construct(
...
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $output->writeln('Uploading assets information from media directory to database...');
-        $this->state->emulateAreaCode(Area::AREA_ADMINHTML, function () {
-            $this->imagesIndexer->execute();
+        $this->state->emulateAreaCode(Area::AREA_ADMINHTML, function () use ($output) {
+            foreach ($this->indexers as $indexer) {
+                /** @var ImagesIndexerInterface $indexer */
+                if ($indexer instanceof ImagesIndexerInterface) {
+                    $output->write($indexer->getTitle() . ' index ');
```
Comment (hws47a) [inline MediaGalleryUi/Console/Command/IndexAssets.php]: Probably `' index '` string should be translatable as well.

File: MediaGalleryUi/Console/Command/IndexAssets.php [hunk 3]
```diff
@@ -51,19 +51,31 @@ public function __construct(
...
+                    $indexer->execute();
+                    $resultTime = microtime(true) - $startTime;
+                    $output->writeln(
+                        __('has been rebuilt successfully in %time', ['time' => gmdate('H:i:s', (int) $resultTime)])
+                    );
+                }
+            }
         });
-        $output->writeln('Completed assets indexing.');
+        $output->writeln('Completed media galley ui indexing.');
```
Comment (hws47a) [inline MediaGalleryUi/Console/Command/IndexAssets.php]: Please wrap the string in `__()`

**Review by hws47a — CHANGES_REQUESTED** [review]
Hi @coderimus,

It's a very nice improvement, please check my minor comments

**Comment by coderimus** [comment]
@hws47a thank you for your review. I implemented the requested changes. Please, review.

File: MediaGalleryUi/Console/Command/IndexAssets.php [hunk 4]
```diff
@@ -51,19 +51,32 @@ public function __construct(
...
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $output->writeln('Uploading assets information from media directory to database...');
-        $this->state->emulateAreaCode(Area::AREA_ADMINHTML, function () {
-            $this->imagesIndexer->execute();
+        $this->state->emulateAreaCode(Area::AREA_ADMINHTML, function () use ($output) {
+            foreach ($this->indexers as $indexer) {
+                /** @var ImagesIndexerInterface $indexer */
+                if ($indexer instanceof ImagesIndexerInterface) {
+                    $indexTitle = __($indexer->getTitle() . ' index ');
```
Comment (hws47a) [inline MediaGalleryUi/Console/Command/IndexAssets.php]:
```suggestion
                    $indexTitle = __('%1 index ', $indexer->getTitle());
```

Comment (coderimus) [reply MediaGalleryUi/Console/Command/IndexAssets.php]: @hws47a thank you for pointing this part of a code. I made adjustments. Please, review :)

**Review by hws47a — DISMISSED** [review]
Everything looks good to me. Thanks for all of your changes.

**Pull request closed without merging**